app/app_test.go

```diff
@@ -261,275 +269,102 @@ func TestDifferentTxForms(t *testing.T) {
-		Fee:          0,
-	}
-
-	// Sign the hash of the transaction
-	tx := GetTx(msg, keys[0])
-
-	cc.BeginBlock(abci.RequestBeginBlock{Header: abci.Header{Height: 7}})
-
-	// Run a check
-	cres := cc.Check(tx)
-	require.Equal(t, sdk.CodeType(0), sdk.CodeType(cres.Code), cres.Log)
-
-	dres := cc.Deliver(tx)
-
-	require.Equal(t, sdk.CodeType(0), sdk.CodeType(dres.Code), dres.Log)
-
-	// Create context
-	ctx := cc.NewContext(false, abci.Header{})
-
-	// Retrieve UTXO from context
-	utxo1 := cc.utxoMapper.GetUTXO(ctx, addrs[1], types.NewPosition(7, 0, 0, 0))
-	expected1 := types.NewBaseUTXO(addrs[1], [2]common.Address{addrs[0], common.Address{}}, 20, types.NewPosition(7, 0, 0, 0))
-	utxo2 := cc.utxoMapper.GetUTXO(ctx, addrs[2], types.NewPosition(7, 0, 1, 0))
-	expected2 := types.NewBaseUTXO(addrs[2], [2]common.Address{addrs[0], common.Address{}}, 80, types.NewPosition(7, 0, 1, 0))
-
-	require.Equal(t, expected1, utxo1, "First UTXO did not get added to store correctly")
-	require.Equal(t, expected2, utxo2, "Second UTXO did not get added to store correctly")
-
-	utxo1 = cc.utxoMapper.GetUTXO(ctx, addrs[0], types.NewPosition(0, 0, 0, 1))
-	require.Nil(t, utxo1, "Deposit was not removed from utxo store")
-
-	cc.EndBlock(abci.RequestEndBlock{})
+	InitTestChain(addrs, cc)
 	cc.Commit()
 
-	// 2 different inputs 1 output
-	// Block 8
-	confirmSig1 = CreateConfirmSig(types.NewPosition(7, 0, 0, 0), keys[0], &ecdsa.PrivateKey{}, false)
-	confirmSig2 := CreateConfirmSig(types.NewPosition(7, 0, 1, 0), keys[0], &ecdsa.PrivateKey{}, false)
-
-	msg = types.SpendMsg{
-		Blknum1:      7,
-		Txindex1:     uint16(0),
-		Oindex1:      uint8(0),
-		DepositNum1:  0,
-		Owner1:       addrs[1],
-		ConfirmSigs1: confirmSig1,
-		Blknum2:      7,
-		Txindex2:     uint16(0),
-		Oindex2:      uint8(1),
-		DepositNum2:  0,
-		Owner2:       addrs[2],
-		ConfirmSigs2: confirmSig2,
-		Newowner1:    addrs[3],
-		Denom1:       100,
-		Newowner2:    common.Address{},
-		Denom2:       0,
-		Fee:          0,
+	cases := []struct {
+		owner_index1   uint64
+		position1      types.Position
+		input_index1_1 uint64
+		input_index1_2 uint64
+		owner_index2   uint64
+		addr2          common.Address
+		position2      types.Position
+		input_index2_1 uint64
+		input_index2_2 uint64
+		newowner1      common.Address
+		denom1         uint64
+		newowner2      common.Address
+		denom2         uint64
+	}{
+		// 1 input 2 output
+		{0, types.NewPosition(0, 0, 0, 1), 0, 0, 0, common.Address{}, types.Position{}, 0, 0, addrs[1], 20, addrs[2], 80},
+		// 2 different inputs, 1 output
+		{1, types.NewPosition(7, 0, 0, 0), 0, 0, 2, addrs[2], types.NewPosition(7, 0, 1, 0), 0, 0, addrs[3], 100, common.Address{}, 0},
+		// 1 input 2 ouput
+		{3, types.NewPosition(8, 0, 0, 0), 1, 2, 0, common.Address{}, types.Position{}, 0, 0, addrs[3], 75, addrs[4], 25},
+		// 2 different inputs 2 outputs
+		{3, types.NewPosition(9, 0, 0, 0), 3, 0, 4, addrs[4], types.NewPosition(9, 0, 1, 0), 3, 0, addrs[3], 70, addrs[3], 30},
```

Comment:
Would be nice to have another newline comment here explaining what is happening in each case. For the above example:

// Tx spends the 2 outputs from testcase 3 and creates 2 new outputs.